src/main/java/org/broadinstitute/hellbender/tools/picard/illumina/ExtractIlluminaBarcodes.java

```diff
@@ -581,55 +520,52 @@ public PerTileBarcodeExtractor(
-                final int[] barcodeIndices = outputReadStructure.barcodes.getIndices();
-                final BufferedWriter writer = openFileForBufferedWriting(barcodeFile);
-                final byte barcodeSubsequences[][] = new byte[barcodeIndices.length][];
-                final byte qualityScores[][] = usingQualityScores ? new byte[barcodeIndices.length][] : null;
-                while (provider.hasNext()) {
-                    // Extract the barcode from the cluster and write it to the file for the tile
-                    final ClusterData cluster = provider.next();
-                    for (int i = 0; i < barcodeIndices.length; i++) {
-                        barcodeSubsequences[i] = cluster.getRead(barcodeIndices[i]).getBases();
-                        if (usingQualityScores) qualityScores[i] = cluster.getRead(barcodeIndices[i]).getQualities();
-                    }
-                    final boolean passingFilter = cluster.isPf();
-                    final BarcodeMatch match = findBestBarcodeAndUpdateMetrics(barcodeSubsequences, qualityScores, passingFilter, metrics, noMatch);
-
-                    final String yOrN = (match.matched ? "Y" : "N");
-
-                    for (final byte[] bc : barcodeSubsequences) {
-                        writer.write(bytesToString(bc));
+                try (final BufferedWriter writer = openFileForBufferedWriting(barcodeFile)) {
+                    final int[] barcodeIndices = outputReadStructure.barcodes.getIndices();
+                    final byte barcodeSubsequences[][] = new byte[barcodeIndices.length][];
+                    final byte qualityScores[][] = usingQualityScores ? new byte[barcodeIndices.length][] : null;
+                    while (provider.hasNext()) {
+                        // Extract the barcode from the cluster and write it to the file for the tile
+                        final ClusterData cluster = provider.next();
+                        for (int i = 0; i < barcodeIndices.length; i++) {
+                            barcodeSubsequences[i] = cluster.getRead(barcodeIndices[i]).getBases();
+                            if (usingQualityScores)
+                                qualityScores[i] = cluster.getRead(barcodeIndices[i]).getQualities();
+                        }
+                        final boolean passingFilter = cluster.isPf();
+                        final BarcodeMatch match = findBestBarcodeAndUpdateMetrics(barcodeSubsequences, qualityScores, passingFilter, metrics, noMatch);
+
+                        final String yOrN = (match.matched ? "Y" : "N");
+
+                        for (final byte[] bc : barcodeSubsequences) {
+                            writer.write(bytesToString(bc));
+                        }
+                        writer.write("\t" + yOrN + "\t" + match.barcode + "\t" + valueOf(match.mismatches) +
```

Comment:
This looks like it's run enough that it should probably be replaced with a format or string builder.